google/cloud/forseti/scanner/audit/kms_rules_engine.py

```diff
@@ -0,0 +1,471 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Rules engine for checking crypto keys configuration."""
+
+from collections import namedtuple
+import datetime
+import threading
+
+from google.cloud.forseti.common.gcp_type import resource_util
+from google.cloud.forseti.common.util import logger, date_time, string_formats
+from google.cloud.forseti.scanner.audit import base_rules_engine as bre
+from google.cloud.forseti.scanner.audit import errors as audit_errors
+
+LOGGER = logger.get_logger(__name__)
+
+VIOLATION_TYPE = 'CRYPTO_KEY_VIOLATION'
+
+# Rule Modes.
+WHITELIST = 'whitelist'
+BLACKLIST = 'blacklist'
+REQUIRED = 'required'
+RULE_MODES = frozenset([WHITELIST, BLACKLIST, REQUIRED])
```

Comment:
You have only implemented the blacklist mode, should exclude the rest in the RULE_MODES until you have it ready.

Reply:
Done